sentry-kotlin-compiler-plugin/src/test/kotlin/io/sentry/compose/JetpackComposeInstrumentationTest.kt

```diff
@@ -0,0 +1,291 @@
+            """
+            package io.sentry.compose
+
+            import androidx.compose.ui.Modifier
+            import androidx.compose.ui.semantics.SemanticsPropertyKey
+            import androidx.compose.ui.semantics.semantics
+
+            // Based on TestTag
+            // https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/semantics/SemanticsProperties.kt;l=166;drc=76bc6975d1b520c545b6f8786ff5c9f0bc22bd1f
+            private val SentryTag = SemanticsPropertyKey<String>(
+                name = "SentryTag",
+                mergePolicy = { parentValue, _ ->
+                    // Never merge SentryTags, to avoid leaking internal test tags to parents.
+                    parentValue
+                }
+            )
+
+            object SentryModifierKt {
+
+                private var callback: (tag: String) -> Unit = {}
+
+                @JvmStatic
+                public fun setCallback(c: (tag: String) -> Unit) {
+                    callback = c
+                }
+
+                @JvmStatic
+                public fun Modifier.sentryModifier(tag: String): Modifier {
+                    callback(tag)
+                    return semantics(
+                        properties = {
+                            this[SentryTag] = tag
+                        }
+                    )
+                }
+            }
+            """.trimIndent()
+        )
+        private val fakeComposeFunction = SourceFile.kotlin(
+            name = "ComposableFunction.kt",
+            contents = """
+            package io.sentry.compose
+            import androidx.compose.ui.Modifier
+            import androidx.compose.ui.platform.testTag
+            import androidx.compose.runtime.Composable
+
+            @Composable
+            fun ComposableFunction(
+                modifier : Modifier = Modifier,
+                text: String
+            ) {
+                // no-op
+            }
+            """.trimIndent()
+        )
+
+        fun compileFile(
+            file: SourceFile,
+            includeFakeSentryModifier: Boolean = true
+        ): KotlinCompilation.Result {
+            val result = KotlinCompilation().apply {
+                sources = if (includeFakeSentryModifier) {
+                    listOf(
+                        fakeSentryModifier,
+                        fakeComposeFunction,
+                        file
+                    )
+                } else {
+                    listOf(fakeComposeFunction, file)
+                }
+                compilerPluginRegistrars = listOf(SentryKotlinCompilerPlugin())
+                commandLineProcessors = listOf(SentryKotlinCompilerPluginCommandLineProcessor())
+                inheritClassPath = true
+                messageOutputStream = System.out // see diagnostics in real time
+            }.compile()
+
+            return result
+        }
+
+        fun execute(
```

Comment:
`l`: maybe a little bit of docs would help to understand better what's this going to do